src/cpp/fastdds/xtypes/serializers/json/dynamic_data_json.cpp

```diff
@@ -776,8 +789,18 @@ ReturnCode_t json_serialize_basic_member(
             if (RETCODE_OK == ret)
             {
                 // Insert UTF-8 converted value
+#ifdef MINGW_COMPILER
+                std::string utf8_value;
+                int size_needed = std::wcstombs(nullptr, value.data(), 0);
+                if (size_needed > 0) 
```

Comment:
Same space problem here
```suggestion
                if (size_needed > 0)
```